### PR TITLE
Adding v2plugin tests

### PR DESCRIPTION
**Pull request by neelimamukiri**
Also making the vagrant boxes all visible without setting any env var but preventing them all from starting using autostart: false. This makes it simpler *hopefully* to use the vagrant boxes.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -14,49 +14,82 @@ build:
 ansible-image:
 	@bash ./scripts/build_image.sh
 
-# Brings up a demo cluster to install Contiv on - by default this is a docker, centos cluster.
-# It can be configured to start a RHEL cluster by setting CONTIV_NODE_OS=rhel7.
-# It can be started with k8s kubeadm install by running with CONTIV_KUBEADM=1.
-cluster: cluster-destroy
-	cd cluster && vagrant up
+# Brings up a demo cluster to install Contiv on with docker, centos.
+cluster-docker:
```
Comment (rhim) [inline Makefile]: please rename these to match existing code/doc:

cluster-docker -> cluster-legacy-swarm, here we install docker, swarm, and contiv
cluster-swarm -> cluster-swarm-mode, here we only install contiv

File: Makefile [hunk 2]
```diff
@@ -14,49 +14,82 @@ build:
...
-	BUILD_VERSION=$(rel_ver) CONTIV_KUBEADM=1 make install-test-kubeadm
+	BUILD_VERSION=$(rel_ver) make cluster-kubeadm
+	BUILD_VERSION=$(rel_ver) make install-test-kubeadm
 
 # demo-swarm brings up a cluster with docker swarm, runs the installer on it, and shows the URL
 # of the demo Contiv Admin Console which was set up
 # BUILD_VERSION must be setup to use a specific build, e.g.
 # export BUILD_VERSION=1.0.0-beta.3
 # Or run make as BUILD_VERSION=1.0.0-beta.3 make demo-k8s
 demo-swarm:
```
Comment (rhim) [inline Makefile]: these demo targets may be renamed as well

File: cluster/Vagrantfile [hunk 3]
```diff
@@ -75,36 +92,65 @@ $1
...
+ip = 50
 
-master_ip = create_etc_hosts(num_nodes, base_ip, start)
-create_cfg_info(num_nodes, node_ips)
+node_names = []
+node_ips = []
+node_orcs = []
+master_ips = []
+
+orchestrators.each do |o|
```
Comment (rhim) [inline cluster/Vagrantfile]: what does this outer loop do? Please add comments. In particular, why it is dealing with multiple orchestrators? I thought we setup a single cluster at a time.

File: cluster/Vagrantfile [hunk 4]
```diff
@@ -4,32 +4,49 @@ require 'rubygems'
...
+ORC_SWARM = 'swarm'.freeze
+ORC_KUBEADM = 'kubeadm'.freeze
+ORC_KUBEADM_IDX = 2
+
+# Global vars
 token = 'd900e1.8a392798f13b33a4'
 node_os = ENV['CONTIV_NODE_OS'] || CENTOS
+orc = ENV['CONTIV_ORC'] || ORC_LEGACY_SWARM
+k8s_ver = ENV['CONTIV_K8S_VERSION'] || DEFAULT_K8S_VERSION
+orc_path = k8s_ver.include?('1.5.') || k8s_ver.include?('1.4.') ? 'k8s1.4/' : 'k8s1.6/'
```
Comment (dseevr) [inline cluster/Vagrantfile]: I would rewrite this as follows so you can raise an exception easily:

```rb
orc_path = case k8s_ver
           when /^1\.[45]\./ then "k8s1.4/"
           when /^1\.6\./    then "k8s1.6/"
           else
             raise "unsupported k8s version: #{k8s_ver}"
           end
```

File: cluster/Vagrantfile [hunk 5]
```diff
@@ -75,36 +92,65 @@ $1
...
+orchestrators.each do |o|
+  # Save the start location in IP array
+  start = node_ips.length
+
+  node_names.push(o + '-master')
+  num_workers.times do |n|
+    node_names.push(o + '-worker' + n.to_s)
+  end
+  master_ips.push(base_ip + ip.to_s)
+  num_nodes.times do |_n|
```
Comment (dseevr) [inline cluster/Vagrantfile]: You can completely omit the `|vars|` if you don't use any of them, e.g., `5.times { puts "hi" }`

File: cluster/Vagrantfile [hunk 6]
```diff
@@ -4,32 +4,49 @@ require 'rubygems'
...
-  master_ip = base_ip + start.to_s
+def create_etc_hosts(node_names, node_ips, o)
+  master_ip = node_ips[0]
   hosts = "127.0.0.1   localhost\n"
   hosts << "#{master_ip}   netmaster\n"
-  hosts = (0..num_nodes).inject(hosts) { |acc, elem| acc << base_ip + "#{elem + start} contiv-node#{elem + 1} \n" }
 
-  etc_file = (ENV['VAGRANT_CWD'] || '.') + '/.etc_hosts'
+  node_names.zip(node_ips).each do |node, ip|
+    hosts = hosts + node + ' ' + ip + '\n'
```
Comment (dseevr) [inline cluster/Vagrantfile]: I would change this to

```rb
hosts << "#{node} #{ip}\n"
```

to be consistent with the string appending above it

File: cluster/Vagrantfile [hunk 7]
```diff
@@ -75,36 +92,65 @@ $1
...
-# begin execution here
+# Begin execution here
 num_nodes = ENV['CONTIV_NODES'] ? ENV['CONTIV_NODES'].to_i : 2
+num_workers = num_nodes - 1
 base_ip = ENV['CONTIV_IP_PREFIX'] || '192.168.2.'
-start = ENV['CONTIV_KUBEADM'] ? 50 : 50 + num_nodes
-name_start = ENV['CONTIV_KUBEADM'] ? 1 : 1 + num_nodes
-node_ips = Array.new(num_nodes) { |n| base_ip + (n + start).to_s }
-node_names = Array.new(num_nodes) { |n| "contiv-node#{n + name_start}" }
+num_orcs = orchestrators.length
```
Comment (dseevr) [inline cluster/Vagrantfile]: This var is only used one time in the file... I'd just use `orchestrators.length` there directly

File: cluster/Vagrantfile [hunk 8]
```diff
@@ -75,36 +92,65 @@ $1
...
+node_orcs = []
+master_ips = []
+
+orchestrators.each do |o|
+  # Save the start location in IP array
+  start = node_ips.length
+
+  node_names.push(o + '-master')
+  num_workers.times do |n|
+    node_names.push(o + '-worker' + n.to_s)
```
Comment (dseevr) [inline cluster/Vagrantfile]: It's generally better to interpolate using `#{}` (which automatically calls `.to_s`) than use `+`

```rb
"#{o}-worker#{n}"
```

File: cluster/Vagrantfile [hunk 9]
```diff
@@ -118,29 +164,36 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
...
+      c.vm.hostname = node_name
       c.vm.network :private_network, ip: node_addr
       c.vm.network :private_network, ip: node_addr, virtualbox__intnet: 'true', auto_config: false
-      if n.zero?
-        c.vm.network 'forwarded_port', guest: 10_000, host: 10_005 + node_addr.split('.')[3].to_i
-      end
+
       c.vm.provider 'virtualbox' do |v|
-        v.memory = ENV['CONTIV_KUBEADM'] && n.zero? ? 2048 : 1024
+        v.memory = node_orc == ORC_KUBEADM && master_ips.include?(node_addr) ? 2048 : 512
```
Comment (dseevr) [inline cluster/Vagrantfile]: Why was this cut in half?

File: cluster/Vagrantfile (hunk 5 above)
Comment (dseevr) [inline cluster/Vagrantfile]: You could also create the octets and iterate over them:

```rb
base_octet = 50
octets = base_octet...(base_octet + num_nodes) # three dots creates a Range object exclusive of the endpoint
ips = octets.map { |o| base_ip + o }

ips.each do |ip|
  # do stuff with ip
end
```

File: cluster/Vagrantfile (hunk 9 above)
Comment (neelimamukiri) [inline cluster/Vagrantfile]: I don't think we need the 1024 for worker nodes. Default is 512.

File: cluster/Vagrantfile (hunk 4 above)
Comment (neelimamukiri) [inline cluster/Vagrantfile]: @dseevr , I think what we have should work for 1.7. Will test and update if required.

Comment (dseevr) [inline cluster/Vagrantfile]: .... this is awesome:

```rb
orc_path = case k8s_ver
           when "1.4.".."1.5." then "k8s1.4/"
           when "1.6.".."1.7."    then "k8s1.6/"
           else
             raise "unsupported k8s version: #{k8s_ver}"
           end
```

Apparently, a Range object in a case automatically calls `#cover?` on the expression being evaluated

**Pull request merged**